bundle/parameters.go

```diff
@@ -23,10 +28,30 @@ func (p *Parameter) AppliesTo(action string) bool {
 }
 
 // Validate a Parameter
-func (p *Parameter) Validate() error {
+func (p *Parameter) Validate(name string, bun Bundle) error {
 	if p.Definition == "" {
 		return errors.New("parameter definition must be provided")
 	}
+
+	// Validate default against definition schema, if exists
+	schema, ok := bun.Definitions[p.Definition]
+	if !ok {
+		return fmt.Errorf("unable to find definition for %s", name)
+	}
+	if schema.Default != nil {
+		var valResult *multierror.Error
+		valErrs, err := schema.Validate(schema.Default)
+		if err != nil {
+			valResult = multierror.Append(valResult, errors.Wrapf(err, "encountered an error validating parameter %s", name))
+		}
+		for _, valErr := range valErrs {
+			valResult = multierror.Append(valResult, fmt.Errorf("encountered validation error for parameter %s: %v", name, valErr.Error))
+		}
+		if valResult != nil {
```

Comment:
I think a safer way to check this, so that doesn't return when the multierror is initialized but empty would be:

```suggestion
		if valResult.ErrorOrNil() != nil {
```